ts/session/apis/snode_api/factories/StoreGroupRequestFactory.ts

```diff
@@ -40,6 +40,13 @@ async function makeGroupMessageSubRequest(
     throw new Error('makeGroupMessageSubRequest: not all messages are for the same destination');
   }
 
+  const allTimestamps = uniqBy(compactedMessages, m => m.createAtNetworkTimestamp);
+  if (allTimestamps.length !== compactedMessages.length) {
+    throw new Error(
```

Comment:
Is this even possible with regular messages? Or is it a control message + visible message?

Reply:
```tsx
createAtNetworkTimestamp: createAtNetworkTimestamp + 1, // we just need it to be different than the one of ourLeavingNotificationMessage
```

This answered my question so you can resolve this comment.